python/dgl/cuda/nccl.py

```diff
@@ -0,0 +1,95 @@
+"""API creating NCCL communicators."""
+
+class UniqueId(object):
+    """ Class for allowing python code to create and communicate NCCL Unique
+        IDs, needed for creating communicators.
+    """
+    def __init__(self, id_str=None):
+        """ Create an object reference the current NCCL unique id.
+        """
+
+    def __str__(self):
+        return "UniqueId"
+
+
+class Communicator(object):
+    """ High-level wrapper for NCCL communication.
+    """
+    def __init__(self, size, rank, unique_id):
+        """ Create a new NCCL communicator.
+
+            Parameters
+            ----------
+            size : int
+                The number of processes in the communicator.
+            rank : int
+                The rank of the current process in the communicator.
+            unique_id : NCCLUniqueId
+                The unique id of the root process (rank=0).
+        """
+
+    def sparse_all_to_all_push(self, idx, value, mode):
+        """ Perform an all-to-all-v operation, where by all processors send out
+            a set of indices and corresponding values.
+
+            Parameters
+            ----------
+            idx : IdArray
+                The 1D set of indices to send to other processors.
```

Comment:
Is this store the indices of the processors?

Reply:
This is the indices corresponding to the values. That is, if we have 2 processors, and processor 0 calls this with idx=[0,1,2,3,4,5] and mode='remainder'.

Then it would send `1,3,5` and the corresponding values to processor 1, and receive any even indices from processor 1.

Reply:
I also expanded the documentation of this function in 
5c805fa to make this more clear.

Reply:
The expanded doc is much clear.